### PR TITLE
treesitter: support multi-line nodes in get_node_text

### PR DESCRIPTION
**Pull request by mlvzk**
vim.treesitter.get_node_text now works with multi-line nodes.

Also lets `#lua-match?` match on multi-line nodes, without breaking existing single-line matches.

### CODE REVIEW

File: runtime/lua/vim/treesitter/query.lua [hunk 1]
```diff
@@ -165,11 +163,21 @@ function M.get_node_text(node, source)
...
+      end_index = end_col - start_col
     end
-    local line = a.nvim_buf_get_lines(source, start_row, start_row+1, true)[1]
-    return string.sub(line, start_col+1, end_col)
+    lines[#lines] = string.sub(lines[#lines], 1, end_index)
+
+    local text = lines[1]
+    for i=2,#lines do
+      text = text .. '\n' .. lines[i]
+    end
```
Comment (theHamsta) [inline runtime/lua/vim/treesitter/query.lua:178]: table.concat?

Comment (mlvzk) [reply runtime/lua/vim/treesitter/query.lua]: Changed, thanks

Comment (theHamsta) [reply runtime/lua/vim/treesitter/query.lua]: Can you also check whether match? keeps working with Multi-Line input? I guess vim regexes are only single line so hopefully it doesn't errror

Comment (mlvzk) [reply runtime/lua/vim/treesitter/query.lua]: #match? doesn't use get_node_text so this PR shouldn't break it. I'll be probably making another PR regarding #match?, I would like it to support multi-line nodes, to make it more consistent with the official tree-sitter rust bindings.

Comment (theHamsta) [reply runtime/lua/vim/treesitter/query.lua]: Lua-match? Supports Multi-Line regexes and I guess it already uses get_nod_text

Comment (mlvzk) [reply runtime/lua/vim/treesitter/query.lua]: lua-match does use get_node_text, but has it's own check that prevents it from working on multi-line nodes. Still works on single-line nodes.

```lua
      local start_row, _, end_row, _ = node:range()
      if start_row ~= end_row then
        return false
      end
```

Comment (mlvzk) [reply runtime/lua/vim/treesitter/query.lua]: After removing the check it still works the same on single-line nodes, and works correctly on multi-line nodes. Should I do commit it in this PR, or make a new PR?

Comment (vigoux) [reply runtime/lua/vim/treesitter/query.lua]: It'll be good to have kn this PR as the changes are related

Comment (mlvzk) [reply runtime/lua/vim/treesitter/query.lua]: commited https://github.com/neovim/neovim/pull/14445/commits/2b02772698b49ea8fb319f9762d38a9b4fde6211

**Comment by vigoux** [comment]
Could you add a little test for this feature ? Not mandatory for the `lua-match?` part, but at least for the multiline `get_node_text` ?

**Comment by mfussenegger** [comment]
Merged https://github.com/neovim/neovim/pull/14999

**Pull request closed without merging**